### PR TITLE
[ISSUE #6456] Fix ConsumeMessageService#updateCorePoolSize: Can not i…

### LINKED ISSUE #6456: Bug in ConsumeMessageService#updateCorePoolSize: Can not increase consumeExecutor coreSize.
**BUG REPORT**

1. Please describe the issue you observed:
ConsumeMessageService#updateCorePoolSize cannot increase consumeExecutor coreSize.because "corePoolSize" input must < "this.defaultMQPushConsumer.getConsumeThreadMax()",but consumeExecutor coreSize and maxSize has same value after init,which means there is no way to  increase coreSize.
- What did you do (The steps to reproduce)?
call method "org.apache.rocketmq.client.impl.consumer.ConsumeMessageConcurrentlyService#updateCorePoolSize"
- What is expected to see?
increase consumeExecutor coreSize
- What did you see instead?
nothing,just return;
2. Please tell us about your environment:
JDK8 AND rocketmq-client:5.1.0
3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):
image
image

image

### PR DESCRIPTION
**Pull request by crzbird**
…ncrease consumeExecutor coreSize.

**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix #6456

## Brief changelog

fix method "updateCorePoolSize" of ConsumeMessageConcurrentlyService&ConsumeMessageOrderlyService&ConsumeMessagePopConcurrentlyService&ConsumeMessagePopOrderlyService.Make it support increase consumeExecutor coreSize.
image

Because of JDK threadPool will terminate thread if workerCount > maximumPoolSize,should update maximumPoolSize at the same time;
image
image

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageConcurrentlyService.java [hunk 1]
```diff
@@ -105,9 +105,11 @@ public void shutdown(long awaitTerminateMillis) {
...
     @Override
     public void updateCorePoolSize(int corePoolSize) {
-        if (corePoolSize > 0
-            && corePoolSize <= Short.MAX_VALUE
-            && corePoolSize < this.defaultMQPushConsumer.getConsumeThreadMax()) {
+        if (corePoolSize > 0 && corePoolSize != this.consumeExecutor.getCorePoolSize()
+                && corePoolSize <= Short.MAX_VALUE) {
+            if (corePoolSize > this.consumeExecutor.getMaximumPoolSize()) {
+                this.consumeExecutor.setMaximumPoolSize(corePoolSize);
+            }
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageConcurrentlyService.java:112]: Need an else here or return early.

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageOrderlyService.java [hunk 2]
```diff
@@ -115,9 +115,11 @@ public synchronized void unlockAllMQ() {
...
     @Override
     public void updateCorePoolSize(int corePoolSize) {
-        if (corePoolSize > 0
-            && corePoolSize <= Short.MAX_VALUE
-            && corePoolSize < this.defaultMQPushConsumer.getConsumeThreadMax()) {
+        if (corePoolSize > 0 && corePoolSize != this.consumeExecutor.getCorePoolSize()
+                && corePoolSize <= Short.MAX_VALUE) {
+            if (corePoolSize > this.consumeExecutor.getMaximumPoolSize()) {
+                this.consumeExecutor.setMaximumPoolSize(corePoolSize);
+            }
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageOrderlyService.java:122]: Same here

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessagePopConcurrentlyService.java [hunk 3]
```diff
@@ -91,9 +91,11 @@ public void shutdown(long awaitTerminateMillis) {
...
     @Override
     public void updateCorePoolSize(int corePoolSize) {
-        if (corePoolSize > 0
-            && corePoolSize <= Short.MAX_VALUE
-            && corePoolSize < this.defaultMQPushConsumer.getConsumeThreadMax()) {
+        if (corePoolSize > 0 && corePoolSize != this.consumeExecutor.getCorePoolSize()
+                && corePoolSize <= Short.MAX_VALUE) {
+            if (corePoolSize > this.consumeExecutor.getMaximumPoolSize()) {
+                this.consumeExecutor.setMaximumPoolSize(corePoolSize);
+            }
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessagePopConcurrentlyService.java:98]: Same here

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessagePopOrderlyService.java [hunk 4]
```diff
@@ -109,9 +109,11 @@ public synchronized void unlockAllMessageQueues() {
...
     @Override
     public void updateCorePoolSize(int corePoolSize) {
-        if (corePoolSize > 0
-            && corePoolSize <= Short.MAX_VALUE
-            && corePoolSize < this.defaultMQPushConsumer.getConsumeThreadMax()) {
+        if (corePoolSize > 0 && corePoolSize != this.consumeExecutor.getCorePoolSize()
+                && corePoolSize <= Short.MAX_VALUE) {
+            if (corePoolSize > this.consumeExecutor.getMaximumPoolSize()) {
+                this.consumeExecutor.setMaximumPoolSize(corePoolSize);
+            }
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessagePopOrderlyService.java:116]: Same here

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageConcurrentlyService.java (hunk 1 above)
Comment (crzbird) [reply client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageConcurrentlyService.java:112]: Do you mean if input corePoolSize > this.consumeExecutor.getMaximumPoolSize() then return , not to update maximumPoolSize?

**Comment by crzbird** [comment]
I recognize this issue cause by rocketmq-spring,@interface RocketMQMessageListener use consumeThreadNumber() as consumeExecutor coreSize and maximumSize, two property has same value cause this issue.I've submit a PR to rocketmq-spring.
By the way ,this PR optimize ConsumeMessageService#updateCorePoolSize as well.

**Review by Oliverwqcwrw — APPROVED** [review]

**Pull request closed without merging**